spec/requests/api/admin/editor_can_publish_article_spec.rb

```diff
@@ -0,0 +1,118 @@
+RSpec.describe 'Api::Admin::Articles :update', type: :request do
+  let!(:article) { create(:article) }
+  let(:editor) { create(:user, role: 'editor') }
+  let(:editors_credentials) { editor.create_new_auth_token }
+  let(:editors_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(editors_credentials) }
+
+  let(:journalist) { create(:user, email: "asd@asd.com", role: 'journalist') }
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(journalist_credentials) }
+
+  describe 'editor successfully updates and' do
+    before do
+      put "/api/admin/articles/#{article.id}", headers: editors_headers, params: { activity: "PUBLISH", premium: true , category: 'economy'}
+      article.reload()
+    end
+
+    it 'gives a success code' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'gives a success message' do
+      expect(response_json['message']).to eq "Article successfully published!"
+    end
+
+    it 'has set published to true' do
+      expect(article.published).to eq true
+    end
+
+    it 'has set category to economy' do
+      expect(article.category).to eq "economy"
```

Comment:
Where did we set it to economy?

Reply:
Line 13 we send category as a param, which is economy, sorry for long line, added line breaks to puts